### PR TITLE
Fix a11y scrolling for reversed lists

### LINKED ISSUE #52032: When ListView is reversed, accessible users cannot reach appbar and bottomNavigationBar
Internal bug: b/150855012

Use case: For a chat app, ListView is reversed to show the latest messages at the bottom. 

Currently when a ListView with vertical axis is reversed, the behavior is:
1. When ListView is at the bottom, users cannot go down to reach input box, but can go up to reach app bar (see internal bug for a video)
2. When ListView is at the top, users can go down to reach input box, but cannot reach app bar (see internal bug for a video)

As a result, users go all the way down to read new messages, but cannot reach input box to reply; when they try to go up and find old messages, they will reach app bar instead (which means old messages are essentially lost).

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text(widget.title)),
      body: ListView.builder(
          reverse: true,
          controller: ScrollController(),
          itemCount: 15,
          itemBuilder: (context, index) {
            return SizedBox(
              height: 50,
              child: Text('This is the #$index chat message'),
            );
          }),
      bottomNavigationBar: Card(
        color: Colors.blue,
        child:
            const SizedBox(height: 50, child: Text('Message input box')),
      ),
    );
  }
}
```

### PR DESCRIPTION
**Pull request by goderbauer**
## Description

Prior to this change the semantics tree contained incorrect information about the available scroll actions for reversed lists. This was breaking a11y focus traversal because when scrolled to the end of a list instead of moving a11y focus outside of the list Android thought there are more items in the list and tried to scroll the list instead to expose those (non-existing) items.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/52032.

## Tests

I added the following tests:

* Available semantic scroll actions (forward, vertical)
* Available semantic scroll actions (reverse, vertical)
* Available semantic scroll actions (forward, horizontal)
* Available semantic scroll actions (reverse, horizontal)

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/test/widgets/list_view_semantics_test.dart [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+              );
+            },
+          ),
+        ),
+      );
+
+      expect(semantics, includesNodeWith(actions: <SemanticsAction>[SemanticsAction.scrollRight]));
+
+      // Jump to the end.
+      controller.jumpTo(10 * 150.0);
```
Comment (justinmc) [inline packages/flutter/test/widgets/list_view_semantics_test.dart]: Nit: Declare constants for 10 and 150?  It's pretty obvious what they are in these short tests, not a big deal.

**Review by justinmc — APPROVED** [review]
LGTM :+1:

File: packages/flutter/test/widgets/list_view_semantics_test.dart (hunk 1 above)
Comment (goderbauer) [reply packages/flutter/test/widgets/list_view_semantics_test.dart]: Done.

**Review by dnfield — APPROVED** [review]
LGTM

**Review by jonahwilliams — APPROVED** [review]
LGTM too

**Comment by goderbauer** [comment]
Tripple LGTM! Thanks!

**Pull request merged**